### PR TITLE
[FTE-328] Add mark as read optimistic action

### PR DESCRIPTION
**Pull request by ablenesi**

### CODE REVIEW

File: AppCore/appcore-api/src/commonMain/kotlin/me/sphere/appcore/usecases/NotificationListUseCaseImpl.kt [hunk 1]
```diff
@@ -29,13 +29,18 @@ internal fun createNotificationListUseCase(
...
                 notification.title,
                 notification.repositoryFullName,
                 notification.subjectId
             )
         },
         logger = logger,
-        connectivityMonitor = connectivityMonitor
+        connectivityMonitor = connectivityMonitor,
+//        databaseUpdateTracking = DatabaseUpdateTracking(
+//            getUpdateHead = TODO("Not sure what should this query do 🤔"),
```
Comment (ablenesi) [inline AppCore/appcore-api/src/commonMain/kotlin/me/sphere/appcore/usecases/NotificationListUseCaseImpl.kt]: @andersio This is the point I don't understand what UpdateHead stands for in this context? 🤔

Comment (andersio) [reply AppCore/appcore-api/src/commonMain/kotlin/me/sphere/appcore/usecases/NotificationListUseCaseImpl.kt]: `updateHead` is the maximum update timestamp across all items eligible for your view/data source. It helps the data source to understand whether the loaded items might have been changed in the database.

If no filtering is necessary, you can feed it with a query like:
```sql
SELECT MAX(updatedAt) FROM Entity;
```

File: Android/app/src/main/java/me/sphere/unicorn/notificationList/NotificationList.kt [hunk 2]
```diff
@@ -156,6 +169,7 @@ private fun NotificationItemPlaceholder(
         textModifier = Modifier
             .padding(bottom = 4.dp)
             .placeholder(visible = true, color = Color.LightGray),
+        markAsRead = {}
```
Comment (kokeroulis) [inline Android/app/src/main/java/me/sphere/unicorn/notificationList/NotificationList.kt:172]: why empty?

File: AppCore/appcore-actors-backend0/src/commonMain/kotlin/me/sphere/appcore/rest/notifications/NotificationMarkAsReadOperationActor.kt [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+                database.notificationQueries.markAsReadOptimistically(
+                    isRead = true,
+                    id = input.notificationId,
+                    updatedAt = Clock.System.now()
+                )
+            }
+            runAsync {
+                val request = HTTPRequest<Unit>(
+                    method = HTTPRequest.Method.PATCH,
+                    resource = API("notifications/threads/${input.notificationId}"),
```
Comment (danspndl) [inline AppCore/appcore-actors-backend0/src/commonMain/kotlin/me/sphere/appcore/rest/notifications/NotificationMarkAsReadOperationActor.kt]:
```suggestion
                    resource = API("/notifications/threads/${input.notificationId}"),
```

**Review by danspndl — APPROVED** [review]

File: Android/app/src/main/java/me/sphere/unicorn/notificationList/NotificationList.kt (hunk 2 above)
Comment (ablenesi) [reply Android/app/src/main/java/me/sphere/unicorn/notificationList/NotificationList.kt:172]: it's a placeholder implementation nothing should happen

File: AppCore/appcore-actors-backend0/src/commonMain/kotlin/me/sphere/appcore/rest/notifications/NotificationMarkAsReadOperationActor.kt (hunk 3 above)
Comment (ablenesi) [reply AppCore/appcore-actors-backend0/src/commonMain/kotlin/me/sphere/appcore/rest/notifications/NotificationMarkAsReadOperationActor.kt]: fixed 85fd27a

**Pull request merged**